File: WEB/backend/app/schemas/insurance.py
"""Insurance schemas."""

from datetime import date, datetime
from pydantic import BaseModel


class InsuranceCreate(BaseModel):
    region: str
    country_code: str
    country_name: str
    provider_code: str
    provider_name: str
    policy_number: str | None = None
    group_number: str | None = None
    member_id: str | None = None
    plan_name: str | None = None
    plan_type: str | None = None
    coverage_start: date | None = None
    coverage_end: date | None = None
    is_primary: bool = False
    is_active: bool = True
    subscriber_name: str | None = None
    subscriber_relationship: str | None = None
    insurance_phone: str | None = None
    notes: str | None = None


class InsuranceUpdate(BaseModel):
    region: str | None = None
    country_code: str | None = None
    country_name: str | None = None
    provider_code: str | None = None
    provider_name: str | None = None
    policy_number: str | None = None
    group_number: str | None = None
    member_id: str | None = None
    plan_name: str | None = None
    plan_type: str | None = None
    coverage_start: date | None = None
    coverage_end: date | None = None
    is_primary: bool | None = None
    is_active: bool | None = None
    subscriber_name: str | None = None
    subscriber_relationship: str | None = None
    insurance_phone: str | None = None
    notes: str | None = None


class InsuranceResponse(BaseModel):
    id: int
    user_id: int
    region: str
    country_code: str
    country_name: str
    provider_code: str
    provider_name: str
    policy_number: str | None = None
    group_number: str | None = None
    member_id: str | None = None
    plan_name: str | None = None
    plan_type: str | None = None
    coverage_start: date | None = None
    coverage_end: date | None = None
    is_primary: bool
    is_active: bool
    subscriber_name: str | None = None
    subscriber_relationship: str | None = None
    insurance_phone: str | None = None
    notes: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: IOS/ALAFIA/Models/Insurance.swift
import Foundation

struct Insurance: Codable, Identifiable {
    let id: Int
    let userId: Int
    let region: String
    let countryCode: String
    let countryName: String
    let providerCode: String
    let providerName: String
    let policyNumber: String?
    let groupNumber: String?
    let memberId: String?
    let planName: String?
    let planType: String?
    let coverageStart: String?
    let coverageEnd: String?
    let isPrimary: Bool
    let isActive: Bool
    let subscriberName: String?
    let subscriberRelationship: String?
    let insurancePhone: String?
    let notes: String?
    let createdAt: String

    enum CodingKeys: String, CodingKey {
        case id, region, notes
        case userId = "user_id"
        case countryCode = "country_code"
        case countryName = "country_name"
        case providerCode = "provider_code"
        case providerName = "provider_name"
        case policyNumber = "policy_number"
        case groupNumber = "group_number"
        case memberId = "member_id"
        case planName = "plan_name"
        case planType = "plan_type"
        case coverageStart = "coverage_start"
        case coverageEnd = "coverage_end"
        case isPrimary = "is_primary"
        case isActive = "is_active"
        case subscriberName = "subscriber_name"
        case subscriberRelationship = "subscriber_relationship"
        case insurancePhone = "insurance_phone"
        case createdAt = "created_at"
    }
}

struct InsuranceCreate: Encodable {
    let region: String
    let countryCode: String
    let countryName: String
    let providerCode: String
    let providerName: String
    var policyNumber: String?
    var groupNumber: String?
    var memberId: String?
    var planName: String?
    var planType: String?
    var coverageStart: String?
    var coverageEnd: String?
    var isPrimary: Bool = false
    var isActive: Bool = true
    var subscriberName: String?
    var subscriberRelationship: String?
    var insurancePhone: String?
    var notes: String?

    enum CodingKeys: String, CodingKey {
        case region, notes
        case countryCode = "country_code"
        case countryName = "country_name"
        case providerCode = "provider_code"
        case providerName = "provider_name"
        case policyNumber = "policy_number"
        case groupNumber = "group_number"
        case memberId = "member_id"
        case planName = "plan_name"
        case planType = "plan_type"
        case coverageStart = "coverage_start"
        case coverageEnd = "coverage_end"
        case isPrimary = "is_primary"
        case isActive = "is_active"
        case subscriberName = "subscriber_name"
        case subscriberRelationship = "subscriber_relationship"
        case insurancePhone = "insurance_phone"
    }
}

struct InsuranceCountry: Codable, Identifiable {
    let code: String
    let name: String
    var id: String { code }
}

struct InsuranceProvider: Codable, Identifiable {
    let code: String
    let name: String
    var id: String { code }
}
